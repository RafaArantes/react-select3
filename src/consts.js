export const DEFAULT_LANG = {
    empty: 'Ничего не найдено.',
    emptyValue: '-',
    error: 'Не удалось получить данные.',
    minLength: 'Введите минимум ${minlength} символа',
    isPending: '...',
}
