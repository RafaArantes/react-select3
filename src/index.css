body {
	max-width: 960px;
  	margin: 0 auto;
  	padding: 20px 0;
  	font-family: sans-serif;
}
