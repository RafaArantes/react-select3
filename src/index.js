export default from './components/Select'
export FetchSelect from './components/FetchSelect'
