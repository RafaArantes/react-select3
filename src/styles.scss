$gray: #cecece !default;
$gray-light: #efefef !default;
$gray-lighter: #f0f0f0 !default;
$brand-primary: #00a7ff !default;
$border-radius-base: 0 !default;
$padding-base: 10px !default;
//
//--------------
$selector: 'PureReactSelect' !default;
//
//--------------
$color-highlighted: $gray-light !default;
$color-placeholder: $gray !default;
$color-selected: $brand-primary !default;
//
//--------------
$select-bg: #fff !default;
$select-border-radius: $border-radius-base !default;
$select-height: 31px !default;
$select-padding-horizontal: $padding-base !default;
//
//
//--------------
@mixin selected {
    background: $color-selected;
    color: #fff;
}

@mixin box {
    border-radius: $select-border-radius;
    border: 1px solid $gray;
    box-shadow: 1px 1px 1px 0 $gray-lighter;
}

@mixin nowrap {
    white-space: nowrap;
    overflow: hidden;
}

//
//--------------
.#{$selector}__container {
    display: inline-block;
    position: relative;
    background: $select-bg;
    z-index: 0;
    cursor: pointer;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
    outline: none;
    > * {
        box-sizing: border-box;
        outline: none;
    }
}

.#{$selector}--open {
    &.#{$selector}--below {
        .#{$selector}__selection {
            border-radius: $select-border-radius $select-border-radius 0 0;
            border-bottom: 0;
        }
        .#{$selector}__dropdown {
            top: 100%;
            border-radius: 0 0 $select-border-radius $select-border-radius;
        }
    }
    &.#{$selector}--above {
        .#{$selector}__selection {
            border-radius: 0 0 $select-border-radius $select-border-radius;
            border-top: 0;
        }
        .#{$selector}__dropdown {
            bottom: 100%;
            border-radius: $select-border-radius $select-border-radius 0 0;
        }
    }
}

.#{$selector}__selection {
    @include nowrap;
    @include box;
    display: block;
    height: $select-height;
    padding: 0 ($select-padding-horizontal + 10) 0 $select-padding-horizontal;
    line-height: $select-height;

    &.#{$selector}--clearable {
        padding-right: ($select-padding-horizontal + 30);
    }
}

.#{$selector}__selection--placeholder {
    color: $color-placeholder;
}

.#{$selector}__selection-arrow {
    display: block;
    position: absolute;
    right: 10px;
    top: ($select-height / 2) - 2.5;
    height: 5px;
    width: 5px;

    i {
        display: block;
        border-color: currentColor rgba(255, 255, 255, 0) rgba(255, 255, 255, 0) rgba(255, 255, 255, 0);
        border-style: solid;
        border-width: 5px 4px 0 4px;
        height: 0;
        width: 0;
    }
}

.#{$selector}__clear-selection {
    display: block;
    position: absolute;
    right: 20px;
    top: ($select-height / 2) - 2;
    line-height: 0;
    width: 10px;
    height: 10px;
}

.#{$selector}__dropdown {
    display: block;
    position: absolute;
    width: 100%;
    background-color: $select-bg;
    @include box;
}

.#{$selector}__options-list {
    display: block;
    list-style: none;
    padding: 0;
    margin: 0;
    li {
        line-height: $select-height;
        @include nowrap;
        padding: 0 $select-padding-horizontal;
    }
}

.#{$selector}__option--highlighted {
    background-color: $color-highlighted;
}

.#{$selector}__option--selected {
    @include selected;
}

.#{$selector}__search-field {
    display: block;
    -webkit-appearance: none;
    height: 31px;
    padding: 0 $select-padding-horizontal;
    width: calc(100% - #{$select-padding-horizontal * 2});
    border: 0;
    border-bottom: 1px solid #ccc;
    margin: 0 auto;
    outline: none;
}

.#{$selector}__status {
    display: block;
    padding: $padding-base*0.3 $select-padding-horizontal;
    color: $color-placeholder;
}
