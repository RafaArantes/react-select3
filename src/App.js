import React, { Component } from 'react'
// import Select from 'react-select3'

import Select, { FetchSelect } from 'react-select3'
import 'react-select3/dist/styles.css'


const dummy = [
  {
    id: 3,
    text: 'three'
  },
  {
    id: 4,
    text: 'four'
  },
  {
    id: 5,
    text: 'five'
  },
  {
    id: 6,
    text: 'six'
  },
  {
    id: 7,
    text: 'seven'
  },
  {
    id: 8,
    text: 'eight'
  },
  {
    id: 9,
    text: 'nine'
  },
  {
    id: 10,
    text: 'ten'
  },
  {
    id: 11,
    text: 'eleven'
  },
  {
    id: 12,
    text: 'twelve'
  },
];

const dummy2 = [
  {
    id: 13,
    text: 'ninety three'
  },
  {
    id: 14,
    text: 'ninety four'
  },
  {
    id: 15,
    text: 'ninety five'
  },
  {
    id: 16,
    text: 'ninety six'
  },
  {
    id: 17,
    text: 'ninety seven'
  },
  {
    id: 18,
    text: 'ninety eight'
  },
  {
    id: 19,
    text: 'ninety nine'
  },
  {
    id: 110,
    text: 'ninety ten'
  },
  {
    id: 111,
    text: 'ninety eleven'
  },
  {
    id: 12,
    text: 'ninety twelve'
  },
];


const ajaxClient = (endpoint) => {
  console.log(endpoint)

  return Promise.resolve(dummy)
}

class App extends Component {
  constructor(props) {
    super(props)

    this.state = {
      value: null,
      searchTerm: '',
      disabled: false,
      selectedOption: null,
      error: false,
      options: dummy
    }
  }

  _onSearchTermChange = ({ target: { value: searchTerm } }) => {
    console.log(searchTerm)
    this.setState({
      searchTerm
    })
  }

  _onSelect = (e) => {
    console.log(e)
    this.setState({ value: e.target.value, selectedOption: e.target.option })
  }

  _resetSelect = () => {
    this.selectRef.clear()
  }

  _setDummyOptions = () => {
    this.selectRef.options = dummy2
  }

  _toggleDisabled = () => {
    this.setState(({ disabled }) => ({
      disabled: !disabled
    }))
  }

  _toggleError = () => {
    this.setState(({ error }) => ({
      error: !error
    }))
  }

  _getSelectRef = (node) => {
    this.selectRef = node
  }

  _formSubmit = e => {
    e.preventDefault()
    const formData = new FormData(e.target)
    console.dir(e.target)
    for (let entry of formData.entries()) {
      console.log(entry)
    }
  }

  render() {
    const { searchTerm, disabled, selectedOption, error } = this.state

    return (
      <div>
        <FetchSelect fetch={{ ajaxClient, once: true }}/>
        <div>
          <p>
            search term: { searchTerm }
          </p>
          <p>
            is disabled: { String(disabled) }
          </p>
          <p>
            selectedOption: { JSON.stringify(selectedOption) }
          </p>
          <p>
            <button onClick={ this._resetSelect }>
              Reset
            </button>
            <button onClick={ this._setDummyOptions }>
              Set dummy options
            </button>
            <button onClick={ this._toggleDisabled }>
              { disabled ? 'Enable' : 'Disable' }
            </button>
            <button onClick={ this._toggleError }>
              Toggle error
            </button>
          </p>
        </div>
        <form action="#" onSubmit={this._formSubmit}>
          <input type="text" name="dummyName"/>
          <input type="text" name="dummyName2"/>
          <Select name="dummyData3" options={ dummy } placeholder="Select me"/>
          <button type="submit">submit</button>
        </form>
      </div>
    )
  }
}

export default App