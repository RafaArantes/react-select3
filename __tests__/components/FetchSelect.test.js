import { FetchSelect as Select } from '../../src'


describe('<FetchSelect>', () => {
  it('should be rendered', () => {
    const component = shallow(<Select />)

    expect(component).toMatchSnapshot()
  })

  it('should fetch options once on mount', () => {
    const clientSpy = sinon.spy(mock, 'ajaxClient')
    const component = mount(
      <Select fetch={{
        ajaxClient: mock.ajaxClient,
        once: true }}/>)

    expect(component).toMatchSnapshot()
    expect(clientSpy.called).toBe(true)
  })
})
