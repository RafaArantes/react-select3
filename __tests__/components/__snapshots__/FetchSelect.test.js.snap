// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FetchSelect> should be rendered 1`] = `
<Select
  allowClear={false}
  closeOnClickOutside={true}
  cssClassNamePrefix="rs3"
  disabled={false}
  layout={
    Object {
      "dropdownVerticalPosition": "below",
      "width": "245px",
    }
  }
  name="reactSelect_1"
  onSearchTermChange={[Function]}
  options={null}
  search={
    Object {
      "minimumResults": undefined,
      "show": true,
      "status": "Minimum 3 to search...",
    }
  }
/>
`;

exports[`<FetchSelect> should fetch options once on mount 1`] = `
<FetchSelect
  fetch={
    Object {
      "ajaxClient": [Function],
      "once": true,
    }
  }
  search={Object {}}
>
  <Select
    allowClear={false}
    closeOnClickOutside={true}
    cssClassNamePrefix="rs3"
    disabled={false}
    layout={
      Object {
        "dropdownVerticalPosition": "below",
        "width": "245px",
      }
    }
    name="reactSelect_1"
    onSearchTermChange={[Function]}
    options={null}
    search={
      Object {
        "minimumResults": undefined,
        "show": false,
        "status": "...",
      }
    }
  >
    <span
      className="rs3__container  rs3--below"
      disabled={false}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="combobox"
      style={
        Object {
          "width": "245px",
        }
      }
      tabIndex="0"
    >
      <select
        className="rs3__select-node"
        name="reactSelect_1"
        readOnly={true}
        tabIndex="-1"
        value=""
      />
      <SelectSelection
        clearable={false}
        onClearSelection={[Function]}
        onKeyDown={[Function]}
        selection={null}
      >
        <span
          className="rs3__selection rs3__selection--placeholder"
        >
          <span
            className="rs3__selection-text"
          />
          <span
            className="rs3__selection-arrow"
            role="presentation"
          >
            <i />
          </span>
        </span>
      </SelectSelection>
      <SelectError />
    </span>
  </Select>
</FetchSelect>
`;
